pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -105,7 +106,7 @@ func (c *ComponentHandler) IsRevisionDiff(mt metav1.Object, curComp *v1alpha2.Co
 		c.Logger.Info(fmt.Sprintf("get old controllerRevision %s error %v, will create new revision", curComp.Status.LatestRevision.Name, err), "componentName", mt.GetName())
 		return true, curComp.Status.LatestRevision.Revision
 	}
-	oldComp, err := UnpackRevisionData(oldRev)
+	oldComp, err := util.UnpackRevisionData(oldRev)
 	if err != nil {
 		c.Logger.Info(fmt.Sprintf("Unmarshal old controllerRevision %s error %v, will create new revision", curComp.Status.LatestRevision.Name, err), "componentName", mt.GetName())
 		return true, oldRev.Revision
```

Comment:
Why not delete the UnpackRevisionData() function at around 121? I noticed that it's test is deleted..